test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -101,4 +95,38 @@ describe('A LockingResourceStore', (): void => {
     expect(lock.release).toHaveBeenCalledTimes(1);
     expect(order).toEqual([ 'acquire', 'release' ]);
   });
+
+  it('releases the lock on the resource when data has been read.', async(): Promise<void> => {
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const drainData = new Promise((resolve): any => {
+      representation.data.on('data', (): any => true);
+      representation.data.prependListener('end', (): any => {
+        order.push('end');
+        resolve();
+      });
+    });
+    await drainData;
+    expect(locker.acquire).toHaveBeenCalledTimes(1);
+    expect(locker.acquire).toHaveBeenLastCalledWith({ path: 'path' });
+    expect(source.getRepresentation).toHaveBeenCalledTimes(1);
+    expect(lock.release).toHaveBeenCalledTimes(1);
+    expect(order).toEqual([ 'acquire', 'getRepresentation', 'end', 'release' ]);
+  });
+
+  it('releases the lock on the resource when readable errors.', async(): Promise<void> => {
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const handleError = new Promise((resolve): any => {
+      representation.data.prependListener('error', (): any => {
+        order.push('error');
+        resolve();
+      });
+    });
+    representation.data.destroy(new Error('Error on the Readable :('));
```

Comment:
I found it hard to find this line in the test, which is where the action happened. Maybe newline above and below?

Reply:
Good test; and we also need one that calls `destroy` without an error.